capture_service/dive_client_cli.cc

```diff
@@ -701,266 +654,396 @@ void TriggerGfxrCapture(Dive::DeviceManager& mgr, const std::string& gfxr_captur
-    while (std::getline(std::cin, input))
+}  // namespace
+
+// Overload for parsing the Command enum from command line flags.
+bool AbslParseFlag(absl::string_view text, Command* command, std::string* error)
+{
+    if (text.empty())
     {
-        if (input == "t")
-        {
-            std::cout << "key t pressed " << std::endl;
-            TriggerCapture(mgr);
-        }
-        else
+        *command = Command::kNone;
+        return true;
+    }
+    for (const auto& [cmd_enum, meta] : GetCommandRegistry())
+    {
+        if (text == meta.name)
         {
-            break;
+            *command = cmd_enum;
+            return true;
         }
-        std::cout << "Press key t+enter to trigger a capture. \nPress other key+enter to exit.";
     }
-
-    return true;
+    *error = absl::StrCat("\n" + GenerateUsageString());
+    return false;
 }
 
-bool DeployGfxrReplay(Dive::DeviceManager& mgr, const std::string& device_serial)
+// Overload for converting the Command enum back to a string.
+std::string AbslUnparseFlag(Command command)
 {
-    auto dev_ret = mgr.SelectDevice(device_serial);
-
-    if (!dev_ret.ok())
+    if (command == Command::kNone)
     {
-        std::cout << "Failed to select device " << dev_ret.status().message() << std::endl;
-        return false;
+        return "";
     }
+    auto& reg = GetCommandRegistry();
+    auto  it = reg.find(command);
+    return (it != reg.end()) ? it->second.name : "unknown";
+}
 
-    auto dev = *dev_ret;
-    auto ret = dev->SetupDevice();
-    if (!ret.ok())
+// Overload for parsing the AppType enum from command line flags.
+bool AbslParseFlag(absl::string_view text, AppType* type, std::string* error)
+{
+    if (text == "vulkan_openxr")
     {
-        std::cout << "Failed to setup device, error: " << ret.message() << std::endl;
-        return false;
+        *type = AppType::kVulkan_OpenXR;
+        return true;
     }
-    // Deploying install/gfxr-replay.apk
-    ret = mgr.DeployReplayApk(device_serial);
-    if (!ret.ok())
+    if (text == "vulkan_non_openxr")
+    {
+        *type = AppType::kVulkan_Non_OpenXR;
+        return true;
+    }
+    if (text == "vulkan_cli_non_openxr")
+    {
+        *type = AppType::kVulkanCLI_Non_OpenXR;
+        return true;
+    }
+    if (text == "gles_openxr")
     {
-        return false;
+        *type = AppType::kGLES_OpenXR;
+        return true;
     }
 
-    return true;
+    *error = "unknown value for --type. Valid values: vulkan_openxr, vulkan_non_openxr, "
+             "vulkan_cli_non_openxr, gles_openxr";
+    return false;
+}
+
+// Overload for converting the AppType enum back to a string.
+std::string AbslUnparseFlag(AppType type)
+{
+    switch (type)
+    {
+    case AppType::kVulkan_OpenXR:
+        return "vulkan_openxr";
+    case AppType::kVulkan_Non_OpenXR:
+        return "vulkan_non_openxr";
+    case AppType::kVulkanCLI_Non_OpenXR:
+        return "vulkan_cli_non_openxr";
+    case AppType::kGLES_OpenXR:
```

Comment:
need to handle the new case introduced by https://github.com/google/dive/pull/561